File: problems/1331/solution.go
package main
import "sort"

func arrayRankTransform(arr []int) []int {
    sorted, ranks, counter, resp := make([]int, len(arr)), make(map[int]int), 1, make([]int, len(arr))
    
    copy(sorted, arr)
    sort.Ints(sorted)
    
    for _, val := range sorted {
        if _, ok := ranks[val]; !ok {
            ranks[val] = counter
            counter++
        }
    }
    
    for idx, val := range arr {
        resp[idx] = ranks[val]
    }
    
    return resp
}

File: problems/1331/solution.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        sorted_set, counter, ranks_dict = sorted(set(arr)), 1, {}
        
        for num in sorted_set:
            ranks_dict[num] = counter
            counter += 1
        
        return [ranks_dict[num] for num in arr]